server/graphql/v2/object/PayoutMethod.ts

```diff
@@ -10,18 +11,43 @@ const PayoutMethod = new GraphQLObjectType({
     id: {
       type: new GraphQLNonNull(GraphQLString),
       resolve: getIdEncodeResolver(IDENTIFIER_TYPES.PAYOUT_METHOD),
+      description: 'Unique identifier for this payout method',
     },
     type: {
       type: PayoutMethodType,
+      description: 'The type of this payout method (usually the payment provider)',
     },
     name: {
       type: GraphQLString,
+      description: 'A friendly name for users to easily find their payout methods',
+      resolve: (payoutMethod, _, req): string => {
+        // Only collective admins can see the name of a payout method
+        if (req.remoteUser?.isAdmin(payoutMethod.CollectiveId)) {
+          return payoutMethod.isSaved;
+        }
+      },
     },
     isSaved: {
       type: GraphQLBoolean,
+      description: 'Whether this payout method has been saved to be used for future payouts',
+      resolve: (payoutMethod, _, req): boolean => {
+        // Only collective admins can see whether a payout method is saved or not
+        if (req.remoteUser?.isAdmin(payoutMethod.CollectiveId)) {
+          return payoutMethod.isSaved;
+        }
+      },
     },
     data: {
       type: GraphQLJSON,
+      description: 'The actual data for this payout method. Content depends on the type.',
+      resolve: (payoutMethod, _, req): object => {
+        if (
+          req.remoteUser?.isAdmin(payoutMethod.CollectiveId) ||
+          getContextPermission(req, PERMISSION_TYPE.SEE_PAYOUT_METHOD_DATA, payoutMethod.id)
```

Comment:
Isn't the `isAdmin` redundant in this case?

Reply:
The context permission is about whitelisting the field for certain contexts (in this case, an expense where you would be an admin of the host). But no matter what the context is, you should always have access to your own payout method's data.

For example when you list the payout methods for a collective you own with the following query, the permission should not depend on the context - you're always allowed to see your own payout methods no matter what the context is.

```gql
loggedInUser {
  payoutMethods {
    data // Should have access, because user is the owner
  }
}
```